tools/serverpod_cli/test/generator/psql/regression_test.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:path/path.dart' as path;
+import 'package:recase/recase.dart';
+import 'package:serverpod_cli/src/generator/psql/legacy_pgsql_generator.dart';
+import 'package:serverpod_cli/src/test_util/builders/class_definition_builder.dart';
+import 'package:serverpod_cli/src/test_util/builders/generator_config_builder.dart';
+import 'package:test/test.dart';
+
+const projectName = 'example_project';
+final config = GeneratorConfigBuilder().withName(projectName).build();
+const generator = LegacyPgsqlCodeGenerator();
+
+void main() {
+  /**
+   * Citizen -> Company -> Town
+   */
+  group('Given classes with relations when generating pgsql', () {
+    var expectedFileName = path.join(
+      'generated',
+      'tables.pgsql',
+    );
+
+    var citizen = 'citizen';
+    var company = 'company';
+    var town = 'town';
+    var entities = [
+      ClassDefinitionBuilder()
+          .withClassName(citizen.sentenceCase)
+          .withFileName(citizen)
+          .withTableName(citizen)
+          .withSimpleField('name', 'String')
+          .withObjectRelationField(company, company.sentenceCase, company)
+          .build(),
+      ClassDefinitionBuilder()
+          .withClassName(company.sentenceCase)
+          .withFileName(company)
+          .withTableName(company)
+          .withSimpleField('name', 'String')
+          .withObjectRelationField(town, town.sentenceCase, town)
+          .build(),
+      ClassDefinitionBuilder()
+          .withClassName(town.sentenceCase)
+          .withFileName(town)
+          .withTableName(town)
+          .withSimpleField('name', 'String')
+          .build()
+    ];
+
+    var codeMap = generator.generateSerializableEntitiesCode(
+      entities: entities,
+      config: config,
+    );
+
+    test('then expected file is created.', () async {
+      expect(codeMap[expectedFileName], isNotNull);
+    });
+
+    test('then file content matches regression data.', () async {
```

Comment:
Comment: This test is not strictly necessary, it was just a nice way to ensure that nothing disappeared from the output when changing the code.